cmd/image/sync/sync.go

```diff
@@ -0,0 +1,137 @@
+// limitations under the License.
+
+package sync
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+type Spec []struct {
+	Source struct {
+		Bucket string `yaml:"bucket"`
+		Cos    string `yaml:"cos"`
+		Object string `yaml:"object"`
+		Plan   string `yaml:"plan"`
+		Region string `yaml:"region"`
+	} `yaml:"source"`
+	Target []struct {
+		Bucket string `yaml:"bucket"`
+		Plan   string `yaml:"plan"`
+		Region string `yaml:"region"`
+	} `yaml:"target"`
+}
+
+const (
+	ServiceType = "cloud-object-storage"
+)
+
+var Cmd = &cobra.Command{
+	Use:   "sync",
+	Short: "Sync images between IBM COS buckets",
+	Long: `Sync images between IBM COS buckets
+pvsadm image sync --help for information
+
+# Set the API key or feed the --api-key commandline argument
+export IBMCLOUD_API_KEY=<IBM_CLOUD_API_KEY>
+
+Examples:
+
+# using spec yaml file
+pvsadm image sync spec.yaml
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var s3Cli *client.S3Client
+		var apikey string = pkg.Options.APIKey
+
+		//Create bluemix client
+		bxCli, err := client.NewClientWithEnv(apikey, pkg.Options.Environment, pkg.Options.Debug)
+
+		if err != nil {
+			return err
+		}
+
+		// Unmashalling yaml file
+		var spec Spec
+		yamlFile, err := ioutil.ReadFile(args[0])
+		if err != nil {
+			klog.Errorf("yamlFile.Get err   #%v ", err)
+			return err
+		}
+		err = yaml.Unmarshal(yamlFile, &spec)
+		if err != nil {
+			klog.Errorf("Unmarshal: %v", err)
+			return err
+		}
+
+		var selectedObjects []string
+		var locationRes bool
+
+		for _, item := range spec {
+
+			// Creating S3 client
+			s3Cli, err = client.NewS3Client(bxCli, item.Source.Cos, item.Source.Region)
+			if err != nil {
+
+				return err
+			}
+			locationRes, _ = s3Cli.CheckBucketLocationConstraint(item.Source.Bucket, item.Source.Region+"-"+item.Source.Plan)
+			if !locationRes {
+				klog.Errorf("Location constraint verification failed for src bucket: %s", item.Source.Bucket)
+				continue
+			}
+			selectedObjects, err = s3Cli.SelectObjects(item.Source.Bucket, item.Source.Object)
+			if err != nil {
+				klog.Errorf("Select error: %v", err)
+				return err
+			}
+
+			fmt.Println("Selected Objects:", selectedObjects)
+
+			for _, targetItem := range item.Target {
+				s3Cli, err = client.NewS3Client(bxCli, item.Source.Cos, targetItem.Region)
+				if err != nil {
+
+					return err
+				}
+				locationRes, _ = s3Cli.CheckBucketLocationConstraint(targetItem.Bucket, targetItem.Region+"-"+targetItem.Plan)
+				if !locationRes {
+					klog.Errorf("Location constraint verification failed for dest bucket: %s", targetItem.Bucket)
+					continue
+				}
+				for _, srcObject := range selectedObjects {
+					err = s3Cli.CopyObjectToBucket(item.Source.Bucket, targetItem.Bucket, srcObject)
+					if err != nil {
+
```

Comment:
Newline can be removed, same at line number 129 and 131

Reply:
Old change, already removed